drivers/SmartThings/zigbee-fan/src/itm-fan-light/init.lua

```diff
@@ -0,0 +1,168 @@
+-- Copyright 2023 philh30
+--
+-- Licensed under the Apache License, Version 2.0 (the "License");
+-- you may not use this file except in compliance with the License.
+-- You may obtain a copy of the License at
+--
+--     http://www.apache.org/licenses/LICENSE-2.0
+--
+-- Unless required by applicable law or agreed to in writing, software
+-- distributed under the License is distributed on an "AS IS" BASIS,
+-- WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+-- See the License for the specific language governing permissions and
+-- limitations under the License.
+
+local st_device = require "st.device"
+local clusters = require "st.zigbee.zcl.clusters"
+local capabilities = require "st.capabilities"
+local FanControl = clusters.FanControl
+local Level = clusters.Level
+local OnOff = clusters.OnOff
+
+local IFL_FINGERPRINTS = {
+  { mfr = "Samsung Electronics", model = "SAMSUNG-ITM-Z-003" },
+}
+
+local is_ifl = function(opts, driver, device)
+  for _, fingerprint in ipairs(IFL_FINGERPRINTS) do
+    if device:get_manufacturer() == fingerprint.mfr and device:get_model() == fingerprint.model then
+      return true
+    end
+  end
+  return false
+end
+
+-- Create child device
+
+local function add_child(driver,parent,profile,child_type)
+    local child_metadata = {
+        type = "EDGE_CHILD",
+        label = string.format("%s %s", parent.label, child_type),
+        profile = profile,
+        parent_device_id = parent.id,
+        parent_assigned_child_key = child_type,
+        vendor_provided_label = string.format("%s %s", parent.label, child_type)
+    }
+    driver:try_create_device(child_metadata)
+end
+
+local function info_changed(driver, device, event, args)
+  if (device.preferences or {}).childLight2 then
```

Comment:
if the user sets this preference to `false`, should it not delete the device?

Also, as written, on join the device will be a multi-component device with a fan component and a light component, when the preference is changed, there will still be the multi-component device but also a separate child device that is just a light. Is the intent that the component and the child device are the same and will mirror each other? there's no endpoint to component mapping or anything like that. as written, the switch component is non-functional

Reply:
Yes, I tried to use it like that.
Is it a problem if I use it like this?
If it is a problem, should it be automatically deleted from the main card (Light) during child device false?

Reply:
Having a component and a child device that control the same functionality is a problem, yes. I believe you should just stick with the multi-component implementation and get rid of the preference (and the separate presentation file). Since this is for one specific device, we should know whether it has a light or not and can make a decision when we integrate it how we want to present it.

Reply:
Is it okay to delete the preference in profiles and make the lighting and fan automatically main card at the same time when onboarding?
Also, is the presentation file the jason file of CANVAS? Do I have to modify this file as well as the profile? Or can I just modify the profile?

Reply:
So, specifically, I would remove 

```
preferences:
  - title: "Child Device"
    name: childLight2
    description: "Create Child Device : Light"
    required: true
    preferenceType: boolean
    definition:
      default: false
```

from `drivers/SmartThings/zigbee-fan/profiles/itm-fan-light.yml` and then remove the preference handling code from the driver.

Then I would `git rm drivers/SmartThings/zigbee-fan/profiles/switch-level.yml`.

Reply:
I tried to modify the init file or config file in seungkwan-choi:main according to your request, but I can not modify it because the edit file is not activated. Please guide me on how to modify it.

Reply:
Files have been modified including init. Please review it.